src/lib/components/InputGroup/InputGroup.jsx

```diff
@@ -0,0 +1,133 @@
+import PropTypes from 'prop-types';
+import React, { useContext } from 'react';
+import { Text } from '../Text';
+import { withGlobalProps } from '../../provider';
+import { classNames } from '../../utils/classNames';
+import { getRootValidationStateClassName } from '../_helpers/getRootValidationStateClassName';
+import { isChildrenEmpty } from '../_helpers/isChildrenEmpty';
+import { resolveContextOrProp } from '../_helpers/resolveContextOrProp';
+import { transferProps } from '../_helpers/transferProps';
+import { FormLayoutContext } from '../FormLayout';
+import { InputGroupContext } from './InputGroupContext';
+import styles from './InputGroup.scss';
+
+export const InputGroup = ({
```

Comment:
<img width="687" alt="image" src="https://user-images.githubusercontent.com/10213870/212653911-ef818cc8-6123-4030-b5ae-cae5ad8e977b.png">

@adamkudrna @adamkudrna Invalid state is always preferred one. Warning is preffered that success. It is ok to me. But what about colors of validation texts? If one of them is warning and one of them is danger, it always shows red color. Is it ok or do we want to have multiple colors?

Reply:
I think this can be closed. The `InputGroup` can have its own validation state other than the fields inside and, at the same time, the individual fields _cannot_. So it's correct if the color of the validation texts follows the state of the group.